cookbooks/bcpc-hadoop/recipes/datanode.rb

```diff
@@ -165,8 +179,14 @@
   not_if { user_exists? "hcat" }
 end
 
-package 'hive-hcatalog' do
-  action :upgrade
+package hwx_pkg_str('hive-hcatalog', node[:bcpc][:hadoop][:distribution][:release]) do
```

Comment:
Wishful thinking that `datanode.rb` would again only install a datanode some day. My test-kitchen branch requires unpacking all of this so that I can test only HDFS, only YARN and only Hive independently. I am inclined to leave this in its current state rather than combining the logic and making a more difficult review when pulling this apart.